docs/src/reference/gremlin-variants.asciidoc

```diff
@@ -1090,10 +1090,13 @@ on the `ConnectionPoolSettings` instance that can be passed to the `GremlinClien
 |Key |Description |Default
 |PoolSize |The size of the connection pool. |4
 |MaxInProcessPerConnection |The maximum number of in-flight requests that can occur on a connection. |32
+|GetOpenConnectionRetries |The number of retries to get an open connection from the pool to submit a request. |4
```

Comment:
NIT: Maybe we could use `GetReconnectionAttempts()` as a name?

Instead of a number, it could return an `IEnumerable<int>`, that represents milliseconds to wait between each attempt.

Reply:
> Maybe we could use GetReconnectionAttempts() as a name?

I also didn't like the name `GetOpenConnectionRetries` that much. `ReconnectionAttempts` sounds like a better name overall. I will probably omit the _Get_ prefix as it's a property and the _Get_ in `GetOpenConnectionRetries` meant the number of retries to _get_ a connection.

> Instead of a number, it could return an IEnumerable<int>, that represents milliseconds to wait between each attempt.

Do you think that users want to configure the individual wait times explicitly instead of just using the default exponential back-off? And wouldn't an `Action<int>` (or `Action<TimeSpan>`) be even more flexible and easier to configure in that case?
If we let them configure an `Action` to compute the wait times, then it should probably be an additional option though as we would still need an option to configure the number of retries in general.

Reply:
It's true that exponential + jitter is what most people wants, my concern is whether we should allow users to configure the base delay, for example: users might want to start from 200ms, 400ms, 800ms and other people in the order of seconds.

Looking at it a little bit more, maybe there should be two settings: "number of attempts" and "base delay".

Reply:
So, we should introduce a constant factor for the exponential wait function? That would look like this:

```c#
attempt => baseDelay * Math.Pow(2, attempt)
```

and with `baseDelay = TimeSpan.FromMilliseconds(100)` they would get your wait times of 200ms, 400ms, 800ms, and so on.
With `baseDelay = TimeSpan.FromSeconds(1)` as the default we would get the current behaviour in this PR.

Reply:
yes, that would be awesome!

Reply:
I just pushed an update that includes these changes.